hack/jenkins/e2e-runner.sh

```diff
@@ -421,7 +425,7 @@ if [[ "${USE_KUBEMARK:-}" == "true" ]]; then
   # exit non-0.
   # TODO: The above comment is no longer accurate. Need to fix this before
   # turning xunit off for the postsubmit tests. See: #28200
-  ./test/kubemark/run-e2e-tests.sh --ginkgo.focus="${KUBEMARK_TESTS:-starting\s30\spods}" "${KUBEMARK_TEST_ARGS:-}" || dump_cluster_logs
+  ./test/kubemark/run-e2e-tests.sh --ginkgo.focus="${KUBEMARK_TESTS:-starting\s30\spods}" "${KUBEMARK_TEST_ARGS:-}" || dump_cluster_logs_and_exit
```

Comment:
@spxtr - I know this will make the whole suite to be red, instead of producing some xunit files, but I think it is way better, that silently ignore errors and marking runs as "passed". We should fix those "xunit" files later.